### PR TITLE
Hide Stock Blocks

### LINKED ISSUE #255: Hide stock blocks
**Is your feature request related to a problem? Please describe.**

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by Sashank999**
Stock Exchange _ TORN - Google Chrome 16-Feb-21 8_36_06 PM

Closes #255 .

### CODE REVIEW

**Comment by Sashank999** [comment]
Closes #255 .

File: torntools/scripts/content/stockexchange/ttStockexchange.js [hunk 1]
```diff
@@ -504,3 +507,86 @@ function showTotalPortfolioValue() {
...
 }
+
+function hideStockBlocks() {
+	if (hide_stock_blocks.length) {
+		for (let hideStockBlock of hide_stock_blocks) {
+			doc.find(`div.stock-main-wrap li.item.item-wrap[data-stock=${hideStockBlock.toLowerCase()}]`).style.display = "none";
+		}
+		let rawHTML =
+			'<div class="info-msg-cont gray border-round m-top10"><div class="info-msg border-round" style="background-color: #627e0d;"><i class="info-icon"></i><div class="delimiter"><div class="msg right-round" style="background-color: #627e0d;color: #627e0d;">Some stock blocks have been disabled by TornTools. Please re-enable them in Settings.</div></div></div></div>';
+		doc.find("div.stock-main-wrap div.title-black.top-round.m-top10").insertAdjacentHTML("beforeBegin", rawHTML);
```
Comment (DeKleineKobini) [inline torntools/scripts/content/stockexchange/ttStockexchange.js:518]: Adding those extra classes in the selectors is something I would avoid. They might seem to add an extra 'safety', but all they do is another point of failure. In this case, `m-top10` is not present while flying. Just use `.stock-main-wrap .title-black`, or `.stock-main-wrap > .title-black` if you really want more safety.

File: torntools/scripts/content/stockexchange/ttStockexchange.js [hunk 2]
```diff
@@ -504,3 +507,86 @@ function showTotalPortfolioValue() {
...
 	}
 }
+
+function hideStockBlocks() {
+	if (hide_stock_blocks.length) {
+		for (let hideStockBlock of hide_stock_blocks) {
+			doc.find(`div.stock-main-wrap li.item.item-wrap[data-stock=${hideStockBlock.toLowerCase()}]`).style.display = "none";
+		}
+		let rawHTML =
+			'<div class="info-msg-cont gray border-round m-top10"><div class="info-msg border-round" style="background-color: #627e0d;"><i class="info-icon"></i><div class="delimiter"><div class="msg right-round" style="background-color: #627e0d;color: #627e0d;">Some stock blocks have been disabled by TornTools. Please re-enable them in Settings.</div></div></div></div>';
```
Comment (DeKleineKobini) [inline torntools/scripts/content/stockexchange/ttStockexchange.js:517]: I would also add some bottom margin

File: torntools/scripts/content/stockexchange/ttStockexchange.js [hunk 3]
```diff
@@ -504,3 +507,86 @@ function showTotalPortfolioValue() {
...
+		doc.find("div.stock-main-wrap div.title-black.top-round.m-top10").insertAdjacentHTML("beforeBegin", rawHTML);
+	}
+}
+
+function hidePortfolio() {
+	let buttonHTML = "<button class='hide-portfolio'>Hide this stock block</button>";
```
Comment (DeKleineKobini) [inline torntools/scripts/content/stockexchange/ttStockexchange.js:523]: I would shorten it to `Hide`. It's on the portfolio, on each individual stock block, people should know on their own imo.

Comment (DeKleineKobini) [reply torntools/scripts/content/stockexchange/ttStockexchange.js:523]: I can also imagine it not going well on mobile if it's this long

File: torntools/changelog.js [hunk 4]
```diff
@@ -1,10 +1,11 @@
 export default {
 	"v5.7.1 - xxxxx, xxth. 2021": {
 		Features: [
-			"Add an arrow for folding Faction Description. - Sashank999",
-			"Show warning when taking Xanax while the user is already at 1000E. - Sashank999",
-		],
-		Fixes: ["Battle stat disappeared from profile when sending user money. - Sashank999"],
+                "Add an arrow for folding Faction Description. - Sashank999"
```
Comment (DeKleineKobini) [inline torntools/changelog.js:4]: `,` missing

**Comment by DeKleineKobini** [comment]
Haven't tested on mobile though, certainly recommend to do that on this one

**Comment by DeKleineKobini** [comment]
I would also add an option for that button in the portfolio to not be there, might be in the way for some

File: torntools/scripts/content/stockexchange/ttStockexchange.js (hunk 3 above)
Comment (Sashank999) [reply torntools/scripts/content/stockexchange/ttStockexchange.js:523]: > I would shorten it to `Hide`. It's on the portfolio, on each individual stock block, people should know on their own imo.

Well, meaningful names may help future developers recognise the usage of a code. We don't document our code so, that might help a bit.

File: torntools/scripts/content/stockexchange/ttStockexchange.js (hunk 2 above)
Comment (Sashank999) [reply torntools/scripts/content/stockexchange/ttStockexchange.js:517]: That is what the class `info-msg msg` does. It already adds some padding so I don't think its necessary.

File: torntools/changelog.js (hunk 4 above)
Comment (Sashank999) [reply torntools/changelog.js]: Changed.

File: torntools/scripts/content/stockexchange/ttStockexchange.js (hunk 1 above)
Comment (Sashank999) [reply torntools/scripts/content/stockexchange/ttStockexchange.js:518]: Changed.

File: torntools/changelog.js [hunk 5]
```diff
@@ -1,10 +1,11 @@
 export default {
 	"v5.7.1 - xxxxx, xxth. 2021": {
 		Features: [
-			"Add an arrow for folding Faction Description. - Sashank999",
-			"Show warning when taking Xanax while the user is already at 1000E. - Sashank999",
-		],
-		Fixes: ["Battle stat disappeared from profile when sending user money. - Sashank999"],
```
Comment (DeKleineKobini) [inline torntools/changelog.js:7]: Why remove this log?

File: torntools/scripts/content/stockexchange/ttStockexchange.js (hunk 2 above)
Comment (DeKleineKobini) [reply torntools/scripts/content/stockexchange/ttStockexchange.js:517]: That's not what I noticed, will test again

File: torntools/scripts/content/stockexchange/ttStockexchange.js (hunk 3 above)
Comment (DeKleineKobini) [reply torntools/scripts/content/stockexchange/ttStockexchange.js:523]: It's about the inner text of the button though: Screenshot

File: torntools/scripts/content/stockexchange/ttStockexchange.js (hunk 2 above)
Comment (DeKleineKobini) [reply torntools/scripts/content/stockexchange/ttStockexchange.js:517]: Okay, might have seem strange behavior before

File: torntools/scripts/content/stockexchange/ttStockexchange.js (hunk 3 above)
Comment (DeKleineKobini) [reply torntools/scripts/content/stockexchange/ttStockexchange.js:523]: Could also happen on PC, but very unlikely (100s of billions in stock)

File: torntools/changelog.js (hunk 5 above)
Comment (Sashank999) [reply torntools/changelog.js:7]: Resolved.

File: torntools/scripts/content/stockexchange/ttStockexchange.js (hunk 3 above)
Comment (Sashank999) [reply torntools/scripts/content/stockexchange/ttStockexchange.js]: Changed.

**Comment by Sashank999** [comment]
> I would also add an option for that button in the portfolio to not be there, might be in the way for some

Added.

**Pull request merged**